File: frontend/src/types/faultCode.ts
/**
 * @file types/faultCode.ts
 * @description 故障代码体系类型定义
 */

/** 故障代码实体 */
export interface FaultCode {
  id: number;
  code: string;
  faultPhenomenon?: string;
  faultCause?: string;
  solution?: string;
  parentId?: number;
  level?: number;
  categoryIds?: string;
  tenantId?: string;
  status: string;
  sortOrder?: number;
  children?: FaultCode[];
  createTime?: string;
  updateTime?: string;
}

/** 创建故障代码请求 */
export interface CreateFaultCodeRequest {
  code: string;
  faultPhenomenon?: string;
  faultCause?: string;
  solution?: string;
  parentId?: number;
  categoryIds?: string;
  sortOrder?: number;
}

/** 更新故障代码请求 */
export interface UpdateFaultCodeRequest extends CreateFaultCodeRequest {
  status?: string;
}

File: backend/src/main/java/com/ams/entity/FaultCode.java
package com.ams.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.List;

@Data
@TableName("fault_code")
public class FaultCode implements Serializable {

    @TableId(type = IdType.AUTO)
    private Long id;

    /** 故障编码 */
    private String code;

    /** 故障现象（level=1） */
    private String faultPhenomenon;

    /** 故障原因（level=2） */
    private String faultCause;

    /** 解决措施（level=3） */
    private String solution;

    /** 父节点ID */
    private Long parentId;

    /** 层级：1-现象，2-原因，3-措施 */
    private Integer level;

    /** 关联分类ID列表（逗号分隔） */
    private String categoryIds;

    /** 租户ID */
    private String tenantId;

    /** 状态：ENABLED/DISABLED */
    private String status;

    /** 排序号 */
    private Integer sortOrder;

    @TableLogic
    private Integer deleted;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    // ── 非持久化辅助字段 ─────────────────────────────────────────────────────────

    /** 子节点列表（树形结构） */
    @TableField(exist = false)
    private List<FaultCode> children;

    /** 父节点名称（展示用） */
    @TableField(exist = false)
    private String parentName;

    /** 父节点编码（展示用） */
    @TableField(exist = false)
    private String parentCode;
}
